scale/recipe/messages/purge_recipe.py

```diff
@@ -68,6 +74,16 @@ def execute(self):
         """
 
         recipe = Recipe.objects.select_related('superseded_recipe').get(id=self.recipe_id)
+
+        # Kick off purge_source_file for source file inputs of the given recipe
+        input_source_files = RecipeInputFile.objects.filter(Q(recipe__root_superseded_recipe=recipe) |
+                                                            Q(recipe__root_superseded_recipe__isnull=True))
+        input_source_files.select_related('input_file')
```

Comment:
I just realized that this (and job logic as well) only works if the jobs and recipes involved are run with a single source file. If multiple source files kick off these jobs/recipes, we will end up purging all of them. We should a note (new issue?) to address this in the future.

Reply:
I opened #1342.  Feel free to weigh in with your thoughts 💯 👌